lib/Doctrine/ORM/Id/AbstractIdGenerator.php

```diff
@@ -40,7 +40,7 @@ abstract class AbstractIdGenerator
      * By default, this method returns FALSE. Generators that have this requirement
      * must override this method and return TRUE.
      *
-     * @return boolean
+     * @return false
```

Comment:
I think it's better to keep `return boolean` to show all possible values.
It can avoid some `return type no compatible with parent` error message too.

Reply:
Can it really? `false` is a boolean, and is not really a subtype of that. We have both Psalm and Phpstan running on the ORM and neither complained…

Reply:
https://phpstan.org/r/bde04f1f-84a8-4ad9-888d-91b617d29e24

Reply:
Ah it's probably that we don't use a strict enough level yet… I will revert this.